gateway/src/shard/builder.rs

```diff
@@ -44,6 +45,7 @@ impl Error for LargeThresholdError {}
 ///
 /// [`ShardBuilder::shard`]: struct.ShardBuilder.html#method.shard
 #[derive(Debug)]
+#[non_exhaustive]
```

Comment:
this can't be negative, we already cover too large, don't see how anything else could be wrong with the id here